iphone/Maps/Classes/CustomViews/NavigationDashboard/MWMNavigationDashboardManager+Entity.mm

```diff
@@ -95,12 +99,49 @@
 
   return result;
 }
+
+NSAttributedString *buildRouteSegmentsLength(NSArray<MWMRoutePoint *> *points) {
+  auto result = [[NSMutableAttributedString alloc] initWithString:@""];
+  auto &frm = GetFramework();
+  for (int i = 1; i < [points count]; i++) {
+    MWMRoutePoint* segmentStart = points[i-1];
+    MWMRoutePoint* segmentEnd = points[i];
+    m2::PointD startPoint(mercator::LonToX(segmentStart.longitude), mercator::LatToY(segmentStart.latitude));
+    platform::Distance distance;
+    double azimut;
+    frm.GetDistanceAndAzimut(startPoint, segmentEnd.latitude, segmentEnd.longitude, 0, distance, azimut);
```

Comment:
Since we are in cpp world here, it can be simplified:
```
auto const distance = platform::Distance::CreateFormatted(
  ms::DistanceOnEarth(segmentStart.latitude, segmentStart.longitude, segmentEnd.latitude, segmentEnd.longitude));
```

Reply:
Fixed using your proposal.